spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/ServiceInstanceChooser.java

```diff
@@ -33,4 +33,12 @@
 	 */
 	ServiceInstance choose(String serviceId);
 
+	/**
+	 * Chooses a ServiceInstance from the LoadBalancer for the specified service and according to the specified hint.
+	 * @param serviceId The service ID to look up the LoadBalancer.
+	 * @param hint to specify the service instance
+	 * @return A ServiceInstance that matches the serviceId.
+	 */
+	ServiceInstance choose(String serviceId, Object hint);
```

Comment:
Change this to a `default` method that calls `choose(serviceId)`. You'll be able to remove the impls in `*Tests` that throw an exception.

Reply:
@spencergibb thank you for the suggestion. i've applied the change